From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 18 Apr 2020 12:36:49 +0200
Subject: tests: Increase timeouts when running in CI

1 second isn't very much to wait for a D-Bus service in a loaded CI.

Origin: upstream, 1.7.3, commit:cbb95e4d643bab5698700892b0d016a43ca2b2d0
---
 tests/test-portals.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/tests/test-portals.c b/tests/test-portals.c
index d76f4ef..bfa5645 100644
--- a/tests/test-portals.c
+++ b/tests/test-portals.c
@@ -119,6 +119,7 @@ global_setup (void)
   GQuark portal_errors G_GNUC_UNUSED;
   static gboolean name_appeared;
   guint watch;
+  guint timeout_mult = 1;
 
   update_data_dirs ();
 
@@ -133,6 +134,9 @@ global_setup (void)
   g_test_dbus_add_service_dir (dbus, services);
   g_test_dbus_up (dbus);
 
+  if (g_getenv ("TEST_IN_CI"))
+    timeout_mult = 10;
+
   /* g_test_dbus_up unsets this, so re-set */
   g_setenv ("XDG_RUNTIME_DIR", outdir, TRUE);
 
@@ -165,7 +169,7 @@ global_setup (void)
   backends = g_subprocess_launcher_spawnv (launcher, argv, &error);
   g_assert_no_error (error);
 
-  name_timeout = g_timeout_add (1000, timeout_cb, "Failed to launch test-backends");
+  name_timeout = g_timeout_add (1000 * timeout_mult, timeout_cb, "Failed to launch test-backends");
 
   while (!name_appeared)
     g_main_context_iteration (NULL, TRUE);
@@ -208,7 +212,7 @@ global_setup (void)
   g_assert_no_error (error);
   g_clear_pointer (&argv0, g_free);
 
-  name_timeout = g_timeout_add (1000, timeout_cb, "Failed to launch xdg-desktop-portal");
+  name_timeout = g_timeout_add (1000 * timeout_mult, timeout_cb, "Failed to launch xdg-desktop-portal");
 
   while (!name_appeared)
     g_main_context_iteration (NULL, TRUE);
@@ -248,7 +252,7 @@ global_setup (void)
   portals = g_subprocess_launcher_spawnv (launcher, argv, &error);
   g_assert_no_error (error);
 
-  name_timeout = g_timeout_add (1000, timeout_cb, "Failed to launch xdg-permission-store");
+  name_timeout = g_timeout_add (1000 * timeout_mult, timeout_cb, "Failed to launch xdg-permission-store");
 
   while (!name_appeared)
     g_main_context_iteration (NULL, TRUE);
